assets/workflowTemplate.yml

```
@@ -1,23 +1,31 @@
-on: workflow_dispatch
-
 name: test
 
+on: workflow_dispatch
+
 env:
   FOUNDRY_PROFILE: ci
-
+  
 jobs:
   check:
+    strategy:
+      fail-fast: false
+      
     name: Foundry project
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           submodules: recursive
 
       - name: Install Foundry
         uses: foundry-rs/foundry-toolchain@v1
         with:
           version: nightly
+          
+      - name: Run Forge build
+        run: forge build
```

It was suggested in the initial issue that we fail if contracts are above the size limit, so that should be added here

```suggestion
        run: forge build --sizes
```

---

if contracts are above a size limit, does that presuppose that `via_ir=true` is set?

---

No

---

> No

fixed and added log grouping for sizes and tests